cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/validation/ValidatingOutputFormatProvider.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.etl.api.validation;
+
+import io.cdap.cdap.api.data.batch.OutputFormatProvider;
+import io.cdap.cdap.etl.api.FailureCollector;
+
+/**
+ * Validating output format provider.
+ */
+public interface ValidatingOutputFormatProvider extends OutputFormatProvider {
+
+  /**
+   * Validates configurations of output format.
+   *
+   * @param collector failure collector to collect validation failures
+   */
+  void validate(FailureCollector collector);
```

Comment:
does this need to take the input schema as an argument?

Reply:
I think it's more extensible if we pass it a FormatContext (same with the methods in the InputFormat). If we realize we need to add something in the future, we can add it to the context without breaking backwards compatibility.